### PR TITLE
backend compatibility and auth intercept

### PR DESCRIPTION
**Pull request by mseltzer94**
@dcalla02 @jclaw 

Changes to match backend and intercept when a user is logged out (it goes to login)

### CODE REVIEW

File: builds/dumbo/scripts/controllers/newListing.js [hunk 1]
```diff
@@ -103,14 +114,16 @@ angular.module('dumboApp')
...
     };
 
-    if(map[$scope.type]) {
-        text = map[$scope.type]($scope.type);
-        $('#newListing h1').text(text + ' Listing');
-    } else {
-        $scope.type = 'other';
-        text = map['other']('other');
+    $scope.init = function(){
+      console.log('CreatePostCtrl');
```
Comment (jclaw) [inline builds/dumbo/scripts/controllers/newListing.js]: take this out?

File: builds/dumbo/scripts/controllers/newListing.js [hunk 2]
```diff
@@ -103,14 +114,16 @@ angular.module('dumboApp')
...
+      if(!map[$scope.type]) {
+        $scope.type = 'MiscListing';
+      }
+      $('#newListing h1').text(text + ' Listing');
```
Comment (jclaw) [inline builds/dumbo/scripts/controllers/newListing.js]: should we do this with angular?

**Comment by jclaw** [comment]
approved

**Comment by mseltzer94** [comment]
@jclaw @dcalla02 Just refactored this into a service (the map was getting large and doesn't really need to be lumped into the controller). Let me know if you guys see anything I missed

**Comment by jclaw** [comment]
yeah that's better. looks good!

**Pull request merged**